src/main.c

```diff
@@ -7,16 +7,19 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(golioth_rd_template, LOG_LEVEL_DBG);
 
-#include <modem/lte_lc.h>
-#include <net/golioth/system_client.h>
-#include <samples/common/net_connect.h>
-#include <zephyr/net/coap.h>
 #include "app_rpc.h"
 #include "app_settings.h"
 #include "app_state.h"
 #include "app_work.h"
-#include "dfu/app_dfu.h"
+#include <golioth/client.h>
+#include <golioth/fw_update.h>
+#include <samples/common/net_connect.h>
+#include <samples/common/sample_credentials.h>
+#include <zephyr/kernel.h>
 
+#ifdef CONFIG_SOC_NRF9160
```

Comment:
This is probably fine to leave as-is since the nRF9160 is the only supported LTE modem at this point, but something to consider if we need to support another LTE modem in the future: is there a more generic "LTE modem" symbol we can use for this?